File: ionic-vue/inventory/src/store/modules/inventory.ts
import { Module, VuexModule, Mutation, Action, getModule } from 'vuex-module-decorators'
import store from '@/store';
import config from '../../config';

export interface Inventory {
  id: string;
  createdTime: string;
  fields: {
    Posted: string;
    "Product Code": string;
  };
}

@Module({
  dynamic: true,
  namespaced: true,
  name: 'inventory',
  store,
})
class InventoryModule extends VuexModule {
  fetching: boolean = false;
  sending: boolean = false;
  inventory: Inventory[] = [];

  @Mutation
  setFetching(fetching: boolean) {
    this.fetching = fetching;
  }

  @Mutation
  setInventory(inventory: Inventory[]) {
    this.inventory = inventory;
  }

  @Action
  async fetchInventory() {
    this.context.commit('setFetching',true)
    const response = await fetch(
      "https://api.airtable.com/v0/appJkRh9E7qNlXOav/Home?offset=0&maxRecords=100&view=Grid%20view",
      {
        headers: {
          Authorization: config.Authorization
        }
      }
    )
    const body = await response.json();
    this.context.commit('setInventory',body.records);
    this.context.commit('setFetching', false);
  }
}

export default getModule(InventoryModule);

File: flutter_app/lib/models/inventory.dart

import 'package:flutter/material.dart';

class Inventory {
  final String id;
  final String createdTime;
  final Map<String, dynamic> fields;

  Inventory({
    @required this.id,
    @required this.createdTime,
    @required this.fields
  });

  factory Inventory.fromJson(Map<String, dynamic> json) {
    return Inventory(
      id: json['id'],
      createdTime: json['createdTime'],
      fields: {
        'Posted': DateTime.tryParse(json['fields']['Posted']),
        'Product Code': json['fields']['Product Code'] ?? ''
      }
    );
  }
}